tests/mtest/src/main/scala/tests/Assertions.scala

```diff
@@ -57,6 +58,15 @@ trait Assertions extends munit.Assertions {
     }
   }
 
+  def assertSimpleLocationOrdering(locations: List[lsp4j.Location]): Unit = {
```

Comment:
So, I just took a step back and tried to make this as simple as possible. In order to actually be able to just leave the new reference suite the way you had it @tgodzik I just grab the locations when I have them, group them by uri, grab all the line starts, and then assert that they are in order for each uri. The alternative to this would be having to make an entirely different way to do the check. I feel like this is the easiest way to do a simple check on the ordering. What do you think?